Fix implicit function warning with strlcpy. (It's linkable on the system but not included due to the POSIX macros.)

--- missing.h.orig	2021-10-24 10:48:56.000000000 -0400
+++ missing.h	2021-10-24 10:50:49.000000000 -0400
@@ -43,8 +43,6 @@
 void *memrchr (const void *a, int c, size_t len);
 #endif
 
-#ifndef HAVE_STRLCPY
 size_t strlcpy (char *dest, const char *src, size_t dest_size);
-#endif
 
 #endif
